scripts/tasks.py

```diff
@@ -509,3 +511,118 @@ def timetool_correct(config):
+        param_samples = BayesianOptimization.random_param_samples(config)
+
+        # Get the first task of the loop sequence
+        task_to_optimize = config.get(task.task_to_optimize)
+
+        # Get the task generating the scores
+        score_task = config.get(task.score_task)
+
+        # Generate the subdir
+        subdir_name = setup.exp + "_init_samples_configs"
+        subdir_path = os.path.join(setup.root_dir, "bayesian_opt", task.tag, subdir_name)
+        os.makedirs(subdir_path)
+
+        # Generate the config files
+        logger.info(f'Generating {n_samples_init} config files.')
+        for i in range(n_samples_init):
+            config_temp = config.copy()
+            # Overwrite the parameters
+            for j in range(len(params_names)):
+                config_temp[task_to_optimize][params_names[j]] = param_samples[i,j]
+
+            # Overwrite the score task tag to change the name of the score output file
+            config_temp[score_task]["tag"] = config_temp[score_task]["tag"] + f"_sample_{i+1}"
+
+            # Write the config file
+            config_file_name = f"{setup.exp}_sample_{i+1}.yaml"
+            config_file_path = os.path.join(subdir_path, config_file_name)
+
+            with open(config_file_path, 'w') as file:
+                yaml.dump(config_temp, file)
+            
+        logger.info('Done!')
+    else:
+        logger.info('The number of config files to generate was not defined!')
+
+def bo_aggregate_init_samples(config):
+    setup = config.setup
+    task = config.bayesian_opt
+    """ Aggregates the scores and parameters of the initial samples of the Bayesian optimization. """
+    n_samples_init = task.n_samples_init
+    # Get the names of the parameters
+    params_ranges_keys = [key for key in config if key.startswith("range_")]
+    params_names = [key.replace("range_", "") for key in params_ranges_keys]
+    n_params = len(params_names)
+    # Get the first task of the loop sequence
+    task_to_optimize = config.get(task.task_to_optimize)
+    # Get the task generating the scores
+    score_task = config.get(task.score_task)
+
+    # Get the score and the parameters of each sample
+    samples_scores = np.empty(shape=(n_samples_init, 1))
+    samples_params = np.empty(shape=(n_samples_init, n_params))
+
+    logger.info('Aggregating the scores and parameters of the initial samples.')
+    subdir_name = setup.exp + "_init_samples_configs"
+    subdir_path = os.path.join(setup.root_dir, "bayesian_opt", task.tag, subdir_name)
+    for i in range(n_samples_init):
+        # Get the parameters in the config file
+        config_file_name = f"{setup.exp}_sample_{i+1}.yaml"
+        config_file_path = os.path.join(subdir_path, config_file_name)
+        with open(config_file_path, 'r') as file:
+            config_data = yaml.safe_load(file)
+        for j in range(n_params):
+            samples_params[i,j] = config_data[task_to_optimize][params_names[j]]
```

Comment:
It seems like these operations are shared in slightly different flavors between multiple tasks and operations. Perhaps this can be broken out into a seperate utility function(s)?

Reply:
I choose to leave the aggregate task as is for now, as doing utils functions requires a lot of arguments to be passed and the readability is not improved.